test/integration_test_helper.rb

```diff
@@ -20,18 +20,63 @@
 end
 
 Selenium::WebDriver::Chrome::Service.driver_path = ENV['TESTDRIVER_PATH'] || Foreman::Util.which('chromedriver', Rails.root.join('node_modules', '.bin'))
-Capybara.register_driver :selenium_chrome do |app|
+
+javascript_driver = ENV.fetch("JS_TEST_DRIVER") { ENV['DEBUG_JS_TEST'] ? :selenium_chrome : :selenium_chrome_headless }.to_sym
+
+def chrome_options
   options = Selenium::WebDriver::Chrome::Options.new
-  options.args << '--disable-gpu'
-  options.args << '--no-sandbox'
   options.args << '--window-size=1024,768'
-  options.args << '--headless' unless ENV['DEBUG_JS_TEST'] == '1'
-  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
+  options.args += ENV.fetch('ADDITIONAL_CHROME_OPTIONS', '').split(';')
+  options
+end
+
+if javascript_driver == :selenium_chrome_remote
+  ShowMeTheCookies.register_adapter(:selenium_chrome_remote, ShowMeTheCookies::SeleniumChrome)
+
+  Capybara.register_driver :selenium_chrome_remote do |app|
+    selenium_remote_host = ENV.fetch('SELENIUM_REMOTE_HOST')
+    selenium_remote_port = ENV.fetch('SELENIUM_REMOTE_PORT', 4444)
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :remote,
+      url: "http://#{selenium_remote_host}:#{selenium_remote_port}/wd/hub",
+      options: chrome_options)
+  end
+elsif javascript_driver == :selenium_chrome_headless
+  options = chrome_options
+  options.args << '--headless'
```

Comment:
If we keep this, there's also a dedicated API:
```suggestion
  options.headless!
```

Reply:
I don't understand, can you please explain in more detail?

Reply:
see https://github.com/theforeman/foreman/pull/9952#issuecomment-2230534786